pom.xml

```diff
@@ -44,187 +44,288 @@
+						<goals>
+							<goal>integration-test</goal>
+							<goal>verify</goal>
+						</goals>
+					</execution>
+				</executions>
 			</plugin>
+			<!-- Generate Java classes for defined types -->
 			<plugin>
 				<groupId>org.apache.uima</groupId>
-				<artifactId>PearPackagingMavenPlugin</artifactId>
+				<artifactId>jcasgen-maven-plugin</artifactId>
+				<version>${uima-version}</version>
+				<executions>
+					<execution>
+						<id>generate-jcas-classes</id>
+						<phase>generate-sources</phase>
+						<goals>
+							<goal>generate</goal>
+						</goals>
+						<configuration>
+							<typeSystemIncludes>
+								<typeSystemInclude>target/generated-sources/ruta/descriptor/com/example/nlp/ContractRutaTypeSystem.xml</typeSystemInclude>
+							</typeSystemIncludes>
+							<typeSystemExcludes>
+								<typeSystemExclude>/**/BasicTypeSystem.xml</typeSystemExclude>
+								<typeSystemExclude>/**/InternalTypeSystem.xml</typeSystemExclude>
+							</typeSystemExcludes>
+							<limitToProject>true</limitToProject>
+						</configuration>
+					</execution>
+				</executions>
 			</plugin>
+			<!-- Include dependencies in runtime scope as libraries -->
 			<plugin>
-				<groupId>org.codehaus.mojo</groupId>
-				<artifactId>exec-maven-plugin</artifactId>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-dependency-plugin</artifactId>
+				<executions>
+					<execution>
+						<id>copy-dependencies</id>
+						<phase>package</phase>
+						<goals>
+							<goal>copy-dependencies</goal>
+						</goals>
+						<configuration>
+							<outputDirectory>${basedir}/target/pearPackaging/lib</outputDirectory>
+							<overWriteSnapshots>true</overWriteSnapshots>
+							<includeScope>runtime</includeScope>
+						</configuration>
+					</execution>
+				</executions>
 			</plugin>
-		</plugins>
-
-		<pluginManagement>
-			<plugins>
-				<plugin>
+			<plugin>
 					<!-- 
 					  - This plugin generates the XML descriptor for the top-level PEAR analysis component. It runs during the 
 					  - "process-classes" phase such that the descriptor is available when the PEAR is actually packaged later
 					  - in the "package" phase.
 					 -->
-					<groupId>org.codehaus.mojo</groupId>
-					<artifactId>exec-maven-plugin</artifactId>
-					<version>1.6.0</version>
-					<executions>
-						<execution>
-							<id>generate-aae</id>
-							<phase>process-classes</phase>
-							<goals>
-								<goal>java</goal>
-							</goals>
-							<configuration>
-								<mainClass>de.averbis.tutorials.GenerateAggregateAnalysisDescriptor</mainClass>
-								<classpathScope>compile</classpathScope>
-								<arguments>
-									<argument>${project.build.directory}/pearPackaging/desc/de/averbis/tutorials/</argument>
-								</arguments>
-							</configuration>
-						</execution>
-					</executions>
-				</plugin>
-					<!-- 
-					  - This plugin picks up all the dependencies of the PEAR and stages them to a temporary folder. This folder 
```

Comment:
I find this documentation that got nixed more informative than the one we have now.